src/components/S/DefaultObject.vue

```diff
@@ -1,20 +1,20 @@
-<script setup>
+<script setup lang="ts">
 import { ref, watch } from 'vue';
 
-import DefaultObject from './DefaultObject.vue';
+import * as DefaultObject from './DefaultObject.vue';
 import DefaultArray from './DefaultArray.vue';
 import DefaultString from './DefaultString.vue';
 import DefaultNumber from './DefaultNumber.vue';
 import DefaultBoolean from './DefaultBoolean.vue';
 
-const props = defineProps({
-  modelValue: Object,
-  definition: Object
-});
+const props = defineProps<{
+  modelValue: {};
```

Comment:
https://betterprogramming.pub/typescripts-record-type-explained-691372b1a449

Reply:
From that article it sounds like is useful for adding conciseness, I'm not sure `Record<string, any>` does that actually.

Reply:
When you put `{}` it means, that the modelValue will always be {} value and would have nothing inside of it. 

If it were a string and you wanted to draw comparison, it would be `modelValue: ''` vs `modelValue: string`

For object, these days, typescript recommends using `Record<string, any>`